allensdk/brain_observatory/ecephys/stimulus_analysis.py

```diff
@@ -0,0 +1,227 @@
+from six import string_types
+import numpy as np
+import pandas as pd
+import scipy.stats as st
+import scipy.ndimage as ndi
+
+from .ecephys_session import EcephysSession
+
+
+class StimulusAnalysis(object):
+    def __init__(self, ecephys_session, **kwargs):
+        """
+        :param ecephys_session: an EcephySession object or path to ece nwb file.
+        """
+        if isinstance(ecephys_session, EcephysSession):
+            self._ecephys_session = ecephys_session
+        elif isinstance(ecephys_session, string_types):
```

Comment:
rather than handling this in __init__, could you make it a classmethod? Also, consider pathlib.Path